packages/ra-ui-materialui/src/list/filter/FilterListItem.spec.tsx

```diff
@@ -140,6 +147,44 @@ describe('<FilterListItem/>', () => {
                 <FilterListItem label="Foo" value={{ foo: 'bar' }} />
             </ListContextProvider>
         );
-        expect(getByText('Foo').parentElement?.dataset.selected).toBe('true');
+        expect(screen.getByText('Foo').parentElement?.dataset.selected).toBe(
+            'true'
+        );
+    });
+
+    it('should allow to customize isSelected and toggleFilter', () => {
+        const { container } = render(<Cumulative />);
+
+        expect(
+            Array.from(
+                container.querySelectorAll<HTMLElement>(
+                    '[data-selected="true"]'
+                )
+            ).map(item => item.textContent)
+        ).toEqual(['News', 'Tutorials']);
+
+        screen.getByText(/"category":.*\[.*"tutorials",.*"news".*\]/);
```

Comment:
This regex seems a bit too permissive IMO. For instance it would still match if there are too many categories selected.
Why not simply copy paste the exact expected string?

Reply:
Unfortunately I didn't manage to match the full string yet. I'll try

Reply:
Done